sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.institutions.controller.getStudentApplicationSummary.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { DataSource, Repository } from "typeorm";
+import {
+  authorizeUserTokenForLocation,
+  BEARER_AUTH_TYPE,
+  createFakeLocation,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+import {
+  createFakeInstitutionLocation,
+  createFakeApplication,
+  saveFakeStudent,
+  createFakeStudentAssessment,
+  createFakeEducationProgramOffering,
+} from "@sims/test-utils";
+import {
+  Application,
+  ApplicationStatus,
+  EducationProgramOffering,
+  Institution,
+  InstitutionLocation,
+  OfferingIntensity,
+  StudentAssessment,
+} from "@sims/sims-db";
+
+describe("StudentInstitutionsController(e2e)-getStudentApplicationSummary", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let collegeC: Institution;
+  let collegeCLocation: InstitutionLocation;
+  let applicationRepo: Repository<Application>;
+  let studentAssessmentRepo: Repository<StudentAssessment>;
+  let institutionLocationRepo: Repository<InstitutionLocation>;
+  let educationProgramOfferingRepo: Repository<EducationProgramOffering>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    // College C.
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeC = institution;
+    collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+    applicationRepo = appDataSource.getRepository(Application);
+    studentAssessmentRepo = dataSource.getRepository(StudentAssessment);
+    educationProgramOfferingRepo = dataSource.getRepository(
+      EducationProgramOffering,
+    );
+    institutionLocationRepo = dataSource.getRepository(InstitutionLocation);
+  });
+
+  it("Should get the student application details as summary when student has a submitted application for the institution (application with location id saved).", async () => {
+    // Arrange
+
+    // Student has a submitted application to the institution.
+    const student = await saveFakeStudent(appDataSource);
```

Comment:
Is it not possible to simplify the E2E data creation using the `saveFakeApplication`?